tests/conftest.py

```diff
@@ -24,47 +22,3 @@
 def backend():
```

Comment:
We don't use this fixture much anyway, I think we could delete this file?
Looks like it's used only in `test_evaluate` and `test_preprocessing`.

Or if it stays, please update `test_compile` to also use it, so we're more consistent?